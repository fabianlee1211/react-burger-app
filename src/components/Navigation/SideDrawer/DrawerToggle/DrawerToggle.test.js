import React from 'react';
import { shallow } from 'enzyme';
import DrawerToggle from './DrawerToggle';

describe('<DrawerToggle />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<DrawerToggle />);
  });

  it('renders DrawerToggle correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has 3 empty div', () => {
    expect(wrapper.children()).toHaveLength(3);
    wrapper.children().forEach(child => {
      expect(child.equals(<div />)).toBeTruthy();
    });
  });

  it('should trigger onClick', () => {
    const clickFn = jest.fn();
    wrapper.setProps({ clicked: clickFn });
    wrapper.find('.DrawerToggle').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});
