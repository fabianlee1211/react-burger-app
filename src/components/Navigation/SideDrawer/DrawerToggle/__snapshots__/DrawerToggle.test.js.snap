// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerToggle /> renders DrawerToggle correctly 1`] = `
<div
  className="DrawerToggle"
>
  <div />
  <div />
  <div />
</div>
`;
